Automattic/sensei
# Add the email list tables

Fixes #6449

### Changes proposed in this Pull Request

* Render two lists of emails filtered by type (student, teacher) using post meta.
* The lists are rendered in their own email settings sub-tab.
* The lists support pagination.

### Testing instructions

* Enable the feature flag: `add_filter( 'sensei_feature_flag_email_customization', '__return_true' );`
* Go to `/wp-admin/edit.php?post_type=sensei_email`
* Create a few email posts.
* To some of them add post meta `sensei_email_type` with the value `student`.
```php
update_post_meta( $post->ID, 'sensei_email_type', 'student' );
```
* To the others add post meta `sensei_email_type` with the value `teacher`.
```php
update_post_meta( $post->ID, 'sensei_email_type', 'teacher' );
```
* Add a description to some of them.
```php
update_post_meta( $post->ID, 'sensei_email_description', 'Welcome Student' );
```
* Go to Sensei LMS -> Settings -> Emails -> Student Emails.
* Make sure you see the posts with the `student` post meta.
* Go to Sensei LMS -> Settings -> Emails -> Teacher Emails.
* Make sure you see the posts with the `teacher` post meta.
* Make sure you see the description.
* Make sure the pagination works.
* Modify a post and make sure the last modified date works.


### New/Updated Hooks

* `sensei_email_list_columns` – Filter the columns that are displayed in the email list.
* `sensei_email_list_row_data` – Filter the row data displayed in the email list.


### Screenshot / Video

![image](https://user-images.githubusercontent.com/1612178/217957167-1013bdf2-c4d9-4d94-ad24-affe0dac21d5.png)
